docs/DesignDocs/CustomAttestation.md

```diff
@@ -0,0 +1,849 @@
+___
+
+Custom Attestation Data Formats for Open Enclave
+=====
+
+This design document proposes a new attestation framework and set of APIs that
+enable developers to use custom formats for their attestation data.  
+
+Motivation
+----------
+
+Currently, Open Enclave provides several APIs that developers can use for
+attestation. The two key functions are `oe_get_report`, which produces an opaque
+blob that is signed by the enclave, and `oe_verify_report`, which can be used to
+verify the generated report. The original purpose of those two APIs were to
+provide a simple, cross-platform way to produce and verify attestation data.
+
+However, for some developers, need more flexibility for their
+attestation requirements. For example, one might want to extend Open Enclave's
+current attestation structures with extra information, such as geolocation
+or a timestamp. Another user might want their enclaves to generate attestation
+data that is in a compatible format with their existing authentication
+infrastructure, such as a JSON Web Token or a X509 certificate. There are also
+users who want to specify their collaterals (information from a second source
+used for verification), instead of using the set of collaterals provided by Open
+Enclave.
+
+Overall, there has been interest in enhancing Open Enclave's APIs to support
+custom attestation formats to enable these scenarios.
+
+Terminology
+-----------
+
+This document uses the following terminology defined here:
+
+- Claims
+  - Claims are statements about a particular subject. They consist of
+    name-value pairs containing the claim name, which is a string, and
+    claim value, which is arbitrary data. Example of claims could be
+    [key="version", value=1] or [key="enclave_id", value=1111].
+- Evidence
+  - This is the data about the enclave that is produced and signed by it.
+    The SGX report would be an example of evidence.
+- Collaterals
```

Comment:
Yeah, I used collateral since that's what we colloquially refer it as, but it makes sense to be consistent with the RATS WG, considering that the other terms are.

Changed collateral -> endorsements.